### PR TITLE
Reservation CRUD/Update all our error message format

### PR DESCRIPTION
**Pull request by chenyirex**

### CODE REVIEW

File: src/routes/reservations.php [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+ * Created by PhpStorm.
+ * User: chenyi
+ * Date: 2018-11-12
+ * Time: 4:58 PM
+ */
+
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+$app->get('/api/reservations/{username}', function (Request $request, Response $response, array $args) {
```
Comment (suiyoubi) [inline src/routes/reservations.php]: should use `/api/reservations/customer/{username}`, the original URL is very confusing eg: `/api/reservations/1` this is more like the reservation id is 1, not the username=1.

Comment (suiyoubi) [reply src/routes/reservations.php]: also please add a brief comments of what the api do. Easy for front-end guy to know what endpoint to use without reading the code. Could as simple as: `Get a customer's all reservation`

File: src/routes/reservations.php [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        return $response->write(json_encode($reservations))->withStatus(200);
+    } catch (PDOException $e) {
+        $db = null;
+        $errorMessage = $e->getMessage();
+        return $response->write(json_encode(['error' => '' . $errorMessage]))->withStatus(500);
+    }
+});
+
+/**
+ * Example POST body
```
Comment (suiyoubi) [inline src/routes/reservations.php]: this is pretty good 👍

File: src/routes/reservations.php [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+ *       "checkin_date": "2018-11-11",
+ *       "checkout_date": "2018-12-12"
+ *      }
+ * id is not required, if passed in, it will create reservation-room relation on reservation 4.
+ * This would be used when we have multiple rooms reserved under one reservation.
+ * A normal behaviour would be passing in the first request without id, then append the id received from response for the
+ * subsequent requests.
+ * if id not passed in, it will create a new reservation every time.
+ */
+$app->post('/api/reservations/create', function (Request $request, Response $response) {
```
[the remark this answers is not in the document]
Comment (chenyirex) [reply src/routes/reservations.php]: too late bruh, even tho slim can map multiple operations on the same url, you have to merge all CRUD into one handler.

**Comment by suiyoubi** [comment]
We need to find a better way to store the sql file.

**Pull request merged**